.github/workflows/deploy.yml

```diff
@@ -47,34 +55,47 @@ jobs:
     runs-on: 'ubuntu-latest'
     steps:
       - name: Check out repository
+        if: inputs.deploy_type == 'stable'
         uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # https://github.com/actions/checkout/releases/tag/v3.5.3
-      - name: Set preview in package.json
+
+      - name: Check out prerelease
+        if: inputs.deploy_type == 'prerelease'
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # https://github.com/actions/checkout/releases/tag/v3.5.3
+        with:
+          ref: 'f-tfc'
+
+      - name: Set prerelease in package.json
+        if: inputs.deploy_type == 'prerelease'
         id: set-preview
         run: |
           ./build/preview.sh
         env:
           LANGUAGE_SERVER_VERSION: ${{ github.event.inputs.langserver }}
+
       - name: Setup Node
         uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # https://github.com/actions/setup-node/releases/tag/v3.6.0
         with:
           node-version-file: '.nvmrc'
+
       - name: Install dependencies
         run: npm ci
         env:
           npm_config_arch: ${{ matrix.npm_config_arch }}
           ls_target: ${{ matrix.ls_target }}
-      - name: Package VSIX
-        run: npm run package -- --pre-release --target=${{ matrix.vsce_target }}
+
+      - name: Package PreRelease VSIX
+        if: inputs.deploy_type == 'prerelease'
+        run: npm run package -- --githubBranch f-tfc --pre-release --target=${{ matrix.vsce_target }}
```

Comment:
Instead of hard-coding `f-tfc` we could use `$GITHUB_REF_NAME` here and that also means we don't need to do anything special for previews and enable branch-based deploys for stable as a nice side effect.